### DISCUSSION TITLE
Typescript useWatch compile error

### DISCUSSION THREAD

**Discussion by tommiCZE** [react-hook-form/react-hook-form#7413] [General] [answered]
Hi,
I am migrating my project to TypeScript but I got stuck on an error related to useWatch. I create example at [codesandbox](https://codesandbox.io/s/smoosh-dust-r2c23?file=/src/MyInputComponent.tsx). 

Problem is in file MyInputComponent.tsx at line 12.

![obrazek](https://user-images.githubusercontent.com/6510071/147878427-e4e9c79f-10ae-42e1-909b-2ffb0039304a.png)

> (property) name: string[]
No overload matches this call.
  Overload 2 of 4, '(props: { name: Path<T>; defaultValue?: PathValue<T, Path<T>> | undefined; control?: Control<T, object> | undefined; disabled?: boolean | undefined; exact?: boolean | undefined; }): PathValue<...>', gave the following error.
    Type 'string[]' is not assignable to type 'Path<T>'.ts(2769)
useWatch.d.ts(9, 5): The expected type comes from property 'name' which is declared here on type '{ name: Path<T>; defaultValue?: PathValue<T, Path<T>> | undefined; control?: Control<T, object> | undefined; disabled?: boolean | undefined; exact?: boolean | undefined; }'

I have tried various modifying of the generic type when calling u usewatch but without success. any idea where the problem might be?

**Comment by felixschorer** [comment, marked as answer]
That's currently not possible without type assertions. https://github.com/react-hook-form/react-hook-form/discussions/7354 would enable you to do that. Until then you'll have to use type assertions.

https://codesandbox.io/s/zen-chaplygin-s6yov?file=/src/MyInputComponent.tsx

**Reply by tommiCZE** [reply]
@felixschorer 
You solved my problem thanks a lot

**Reply by felixschorer** [reply]
Hm, actually #7354 won't resolve that. At least not in its current form.

**Comment by martin-wahlberg** [comment]
You can make it work without assertions by doing it this way:
https://codesandbox.io/s/react-hook-from-usewatch-forked-ittvj8?file=/src/index.tsx
